controller/controller_test.go

```diff
@@ -312,6 +312,70 @@ func TestController_Start(t *testing.T) {
 	mock.AssertExpectationsForObjects(t, fleetMock.Mock)
 }
 
+
+func TestController_Destroy(t *testing.T) {
+	RegisterTestingT(t)
+
+	// Mocks
+	controller, fleetMock := GivenController()
+	fleetMock.On("GetStatusWithMatcher", mock.AnythingOfType("func(string) bool")).Return(
+		[]fleet.UnitStatus{
+			{
+				Name: "test-main@1.service",
+			},
+			{
+				Name: "test-sidekick@1.service",
+			},
+		},
+		nil,
+	).Once()
+	fleetMock.On("Destroy", "test-main@1.service").Return(nil).Once()
+	fleetMock.On("Destroy", "test-sidekick@1.service").Return(nil).Once()
+
+	// Execute test
+	req := Request{
+		Group:    "test",
+		SliceIDs: []string{"1"},
```

Comment:
Discussed. The actual reason why there is no unit name required and why no `Unit` is given, is the fact that `Destroy` and certain calls do not require the unit name at all. The commands lookup the groups and fetch unit names by the given slice ids to operate on what they found.